File: Compare_N_Contents_Two_Strings/Compare.py
"""
    Author : Pakshal Shashikant Jain
    Date : 15/09/2023
    Program : Write a program which 2 strings from user and check whether first
              N contents of two strings are equal or not. (Implement strcmp()
              function).

              Input : "Marvellous Infosystems"
                      "Marvellous Logic Building"
                      10
              Output : TRUE
"""

##################################################################################################################################

#Function To Compare N Contents of Two String
def StrNCmpX(arr,brr,icnt) :
    i = 0;
    j = 0;

    if((arr == None)and(brr == None)) :
        return False;

    #Logic To Compare
    while(((i != len(arr) - 1)and(j != len(brr)- 1))and(icnt != 0)) :
        if(arr[i] != brr[j]) :
            break;
        i = i + 1;
        j = j + 1;
        icnt = icnt - 1;

    if(arr[i] != brr[j]) :
        return False;
    else :
        return True;

###########################################################################################################################
    
#Main Function
def main() :
    bret = False;
    str = " ";
    str2 = " ";
    icnt = 0;

    print("Jay Ganesh....");
    
    #Taking Input from user
    print("Enter First String");
    str = input();

    print("Enter Second String");
    str2 = input();

    print("Enter One Size");
    icnt = int(input());

    #Call To Behaviours of Class
    bret = StrNCmpX(str,str2,icnt);
    if(bret == True) :
        print("First N Contents Of String Are Equal");
    else :
        print("First N Contents of String Are Not Equal");

####################################################################################################################################

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();

File: Compare_N_Contents_Two_Strings/Compare.cpp
/*
    Author : Pakshal Shashikant Jain
    Date : 15/09/2023
    Program : Write a program which 2 strings from user and check whether first
              N contents of two strings are equal or not. (Implement strcmp()
              function).

              Input : "Marvellous Infosystems"
                      "Marvellous Logic Building"
                      10
              Output : TRUE
*/

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required libraries
#include<iostream>
using namespace std;

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class StringCmp
{
    //Characteristics of Class
    private :
        char *src;
        char *dest;
        int icnt;

    //Constructor of Class
    public : 
        StringCmp(char *SRC,char *DEST,int iCNT)
        {
            src = SRC;
            dest = DEST;
            this->icnt = iCNT;
        }

    //Behaviour of Class To Compare N Contents of Strings
    bool StrNCmpX()
    {
        if((src == NULL)&&(dest == NULL))
        {
            return -1;
        }

        //Logic To Compare
        while(((*src != '\0')&&(*dest != '\0'))&&(this->icnt != 0))
        {
            if(*src != *dest)
            {
                break;
            }
            src++;
            dest++;
            this->icnt--;
        }
        if(*src != *dest)
        {
            return false;
        }
        else
        {
            return true;
        }
    }
};//End of Class

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
int main()
{
    char arr[50] = {'\0'};
    char brr[50] = {'\0'};
    int icnt = 0;
    bool bret = false;

    cout<<"Jay Ganesh...\n";

    //Taking Input from user
    cout<<"Enter First String :\n";
    scanf("%[^'\n']s",arr);

    cout<<"Enter Second String :\n";
    scanf(" %[^'\n']s",brr);

    cout<<"Enter One Size\n";
    cin>>icnt;

    //Creating object of StringCmp Class
    StringCmp *cobj = new StringCmp(arr,brr,icnt);

    //Call To Behaviours of Class
    bret = cobj->StrNCmpX();

    if(bret == true)
    {
        cout<<"First N Contents of String are Equal";
    }
    else
    {
        cout<<"First N Contents of String are Not Equal";
    }

    //Deallocating Memory of Object Created
    delete cobj;

    return 0;
}